spec/classes/profile/haproxy/keepalived_spec.rb

```diff
@@ -22,19 +22,14 @@
       end
 
       let(:thisnode) { { 'ip' => facts[:networking][:ip], 'hostname' => facts[:hostname] } }
-      let(:scotch) { { 'ip' => Faker::Internet.ip_v4_address, 'hostname' => 'scotch' } }
-      let(:soda)   { { 'ip' => Faker::Internet.ip_v4_address, 'hostname' => 'soda' } }
-      let(:coffee) { { 'ip' => Faker::Internet.ip_v4_address, 'hostname' => 'coffee' } }
+      let(:haproxy2) { { 'ip' => Faker::Internet.ip_v4_address, 'hostname' => 'haproxy2' } }
       let(:base_file) { '/etc/keepalived/keepalived.conf' }
       let(:service) { 'keepalived' }
 
-      include_context 'with mocked puppetdb functions', 'somedc', %w[thisnode scotch soda coffee]
+      include_context 'with mocked puppetdb functions', 'somedc', %w[thisnode haproxy2]
 
-      before(:each) do
-        stub('balanced_frontends') do |d|
-          allow_call(d).and_return('www-lib': %w[scotch soda], 'svc2': %w[scotch coffee])
-        end
-      end
+      # required for included haproxy role, but isn't tested here
+      before(:each) { stub('balanced_frontends') { |d| allow_call(d).and_return({}) } }
```

Comment:
I've created https://tools.lib.umich.edu/jira/browse/AEIM-1328, so this is fine as it is, in this scope.